cluster-autoscaler/utils/drain/drain.go

```diff
@@ -74,188 +70,44 @@ const (
 	UnexpectedError
 )
 
-// GetPodsForDeletionOnNodeDrain returns pods that should be deleted on node drain as well as some extra information
-// about possibly problematic pods (unreplicated and DaemonSets).
+// GetPodsForDeletionOnNodeDrain returns pods that should be deleted on node
```

Comment:
I didn't do this because it's a public function and may have downstream dependencies. But since you think it's a good idea, I merged the logic and migrated the tests.